prow/plugins/approve/approve_test.go

```diff
@@ -565,14 +605,307 @@ Approvers can cancel approval by writing `+"`/approve cancel`"+` in a comment
 <!-- META={"approvers":["alice"]} -->`),
 				newTestCommentTime(time.Now(), "alice", "stuff\n/lgtm\nblah"),
 			},
-			selfApprove:       false,
-			needsIssue:        false,
-			lgtmActsAsApprove: false,
+			reviews:             []github.Review{},
+			selfApprove:         false,
+			needsIssue:          false,
+			lgtmActsAsApprove:   false,
+			reviewActsAsApprove: false,
 
 			expectDelete:  false,
 			expectToggle:  false,
 			expectComment: false,
 		},
+		{
+			name:                "approve in review body with empty state",
+			hasLabel:            false,
+			files:               []string{"a/a.go"},
+			comments:            []github.IssueComment{},
+			reviews:             []github.Review{newTestReview("Alice", "stuff\n/approve", "")},
+			selfApprove:         false,
+			needsIssue:          false,
+			lgtmActsAsApprove:   false,
+			reviewActsAsApprove: false,
+
+			expectDelete:  false,
+			expectToggle:  true,
+			expectComment: true,
+			expectedComment: `[APPROVALNOTIFIER] This PR is **APPROVED**
```

Comment:
@cjwagner these tests, validating the _exact_ response content are too specific I suspect :-P

Reply:
They're especially tricky because one of the comment lines includes a trailing space. I had to disable my editor's trailing space cleanup when editing tests.